### DISCUSSION TITLE
screw up obj files indices and vetices when trying to index an unindexed one

### DISCUSSION THREAD

**Discussion by Abdelilah-Majid** [zeux/meshoptimizer#372] [Q&A] [answered]
this is my code :

`

 for (int i = 0; i < g_model.model.meshCount; i++) {

    size_t index_count = g_model.model.meshes[i].triangleCount * 3;

    unsigned int *indices = malloc(sizeof(unsigned int) * index_count);
    g_model.model.meshes[i].indices = (unsigned short *)malloc(sizeof(unsigned short) * index_count);

    unsigned int *remap = malloc(sizeof(unsigned int) * index_count); // allocate temporary memory for the remap table
    size_t vertex_count = meshopt_generateVertexRemap(remap, NULL, index_count, g_model.model.meshes[i].vertices, index_count, sizeof(float));

    float *vertices = malloc(sizeof(float) * vertex_count);

    meshopt_remapIndexBuffer(indices, NULL, index_count, remap);

    meshopt_remapVertexBuffer(vertices, g_model.model.meshes[i].vertices, index_count, sizeof(float), remap);

    for (unsigned short j = 0; j < index_count; j++)
    {
    g_model.model.meshes[i].indices[j] = (unsigned short)indices[j];
    printf("indices_%d: %d\n",j,indices[j]);
    }

    g_model.model.meshes[i].vertices = vertices;
  }

`

and this what happend: 

![Screenshot from 2021-11-21 23-33-13](https://user-images.githubusercontent.com/50689412/142781584-587ad88a-3440-4ea4-a2ae-5a752cd324e9.png)
![Screenshot from 2021-11-21 23-33-29](https://user-images.githubusercontent.com/50689412/142781586-cb78b55a-3777-4212-b3c7-b9e7ad879789.png)

i am using the raylib library and i load obj files that are not indexed
and also i am using fedora linux 64bit system

**Comment by zeux** [comment, marked as answer]
Please refer to something like this as an example: https://github.com/zeux/meshoptimizer/blob/master/demo/main.cpp#L128-L136

In your code one thing that stands out immediately is that you're passing sizeof(float) as vertex size, whereas instead you should be passing the size of the entire vertex - which is typically at least 12 bytes (3 floats), but here likely to be more.
